### ISSUE TITLE
IBAN/BIC  Provider needed

### ISSUE THREAD

**Issue by cdaller** [joke2k/faker#643]
Has someone written an IBAN and BIC provider?
Any other people interested?

**Comment by kungfu71186** [comment]
Not really interested in the data, but I can see it being beneficial. 

https://en.wikipedia.org/wiki/International_Bank_Account_Number
The IBAN consists of up to 34 alphanumeric characters, as follows:
country code using ISO 3166-1 alpha-2 – two letters,
check digits – two digits, and
Basic Bank Account Number (BBAN) – up to 30 alphanumeric characters that are country-specific.[1]

The preferred algorithm is:[8]
Check that the total IBAN length is correct as per the country. If not, the IBAN is invalid.
Replace the two check digits by 00 (e.g., GB00 for the UK).
Move the four initial characters to the end of the string.
Replace the letters in the string with digits, expanding the string as necessary, such that A or a = 10, B or b = 11, and Z or z = 35. Each alphabetic character is therefore replaced by 2 digits
Convert the string to an integer (i.e. ignore leading zeroes).
Calculate mod-97 of the new number, which results in the remainder.
Subtract the remainder from 98, and use the result for the two check digits. If the result is a single digit number, pad it with a leading 0 to make a two-digit number.

Basically IBAN is the same for all countries. BICs are different obviously and are defined per country. Should be fairly straightforward.

**Comment by fcurella** [comment]
That's a good idea, but I don't have enough time to dedicate to the implementation. Feel free to submit a PR and I'll review it :)

**Comment by cdaller** [comment]
PR is on the way :-)

**Comment by mdxs** [comment]
Guess this issue is closed, as the PR has been merged.

**Comment by Menda** [comment]
I just wanted to comment that would be good to have BIC as well.
